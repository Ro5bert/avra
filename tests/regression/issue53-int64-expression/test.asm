.device atmega168

.equ F_CPU = 16000000

.macro delay1
	.set cycles = @0 * (F_CPU / 1000000)
	.dw cycles
	.if cycles != 16000
	.error "error in calculation"
	.endif
.endmacro

.macro delay2
	.set cycles = (@0 * F_CPU) / 1000000
	.dw cycles
	.if cycles != 16000
	.error "error in calculation"
	.endif
.endmacro

.macro delay3
	.set cycles = ( ( @0 * F_CPU ) / 1000000 )
	.dw cycles
	.if cycles != 16000
	.error "error in calculation"
	.endif
.endmacro

nop
delay1 1000
delay2 1000
delay3 1000
