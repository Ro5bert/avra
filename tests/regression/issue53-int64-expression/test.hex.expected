:020000020000FC
:080000000000803E803E803EBE
:00000001FF
